core/coreapi/coreapi.go

```diff
@@ -87,8 +131,110 @@ func (api *CoreAPI) PubSub() coreiface.PubSubAPI {
 	return (*PubSubAPI)(api)
 }
 
+// WithOptions returns api with global options applied
+func (api *CoreAPI) WithOptions(opts ...options.ApiOption) (coreiface.CoreAPI, error) {
+	settings, err := options.ApiOptions(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	if api.nd == nil {
+		return nil, errors.New("cannot apply options to api without node")
+	}
+
+	n := api.nd
+
+	subApi := &CoreAPI{
+		nctx: n.Context(),
+
+		identity:   n.Identity,
+		privateKey: n.PrivateKey,
+
+		repo:       n.Repo,
+		blockstore: n.Blockstore,
+		baseBlocks: n.BaseBlocks,
+		pinning:    n.Pinning,
+
+		blocks: n.Blocks,
+		dag:    n.DAG,
+
+		peerstore:       n.Peerstore,
+		peerHost:        n.PeerHost,
+		namesys:         n.Namesys,
+		recordValidator: n.RecordValidator,
+		exchange:        n.Exchange,
+
+		pubSub: n.PubSub,
+
+		nd: n,
```

Comment:
I'm a *bit* concerned users will see this and think these API's have a parent-child relationship. In practice, options from the parent won't apply to the children. *Should* that happen?

Reply:
Yeah, that's a slight problem. Currently we only have 1 boolean option, so not so much, but once we add more, we should carry options struct, and apply `WithOptions` to that instead of the defaults.